translator/trace/zipkin/status_code_test.go

```diff
@@ -93,3 +93,147 @@ func TestAttribToStatusCode(t *testing.T) {
 		})
 	}
 }
+
+func TestStatusCodeMapperCases(t *testing.T) {
+	tests := []struct {
+		name       string
+		expected   *tracepb.Status
+		attributes map[string]string
+	}{
+		{
+			name:     "no relevant attributes",
+			expected: nil,
+			attributes: map[string]string{
+				"not.relevant": "data",
+			},
+		},
+
+		{
+			name:     "http: 500",
+			expected: &tracepb.Status{Code: 13},
+			attributes: map[string]string{
+				"http.status_code": "500",
+			},
+		},
+
+		{
+			name:     "http: 500",
+			expected: &tracepb.Status{Code: 13, Message: "a message"},
+			attributes: map[string]string{
+				"http.status_code":    "500",
+				"http.status_message": "a message",
+			},
+		},
+
+		{
+			name:     "http: 500, with error attribute",
+			expected: &tracepb.Status{Code: 13},
+			attributes: map[string]string{
+				"http.status_code": "500",
+				"error":            "an error occurred",
+			},
+		},
+
+		{
+			name:     "oc: internal",
+			expected: &tracepb.Status{Code: 13, Message: "a description"},
+			attributes: map[string]string{
+				"census.status_code":        "13",
+				"census.status_description": "a description",
+			},
+		},
+
+		{
+			name:     "oc: description and error",
+			expected: &tracepb.Status{Code: 13, Message: "a description"},
+			attributes: map[string]string{
+				"opencensus.status_description": "a description",
+				"error":                         "INTERNAL",
+			},
+		},
+
+		{
+			name:     "oc: error only",
+			expected: &tracepb.Status{Code: 13, Message: ""},
+			attributes: map[string]string{
+				"error": "INTERNAL",
+			},
+		},
+
+		{
+			name:     "oc: description only, status ok",
+			expected: &tracepb.Status{Code: 0, Message: "a description"},
+			attributes: map[string]string{
+				"opencensus.status_description": "a description",
+			},
+		},
+
+		{
+			name:     "oc: priority over http",
+			expected: &tracepb.Status{Code: 4, Message: "deadline expired"},
+			attributes: map[string]string{
+				"census.status_description": "deadline expired",
+				"census.status_code":        "4",
+
+				"http.status_message": "a description",
+				"http.status_code":    "500",
+			},
+		},
+
+		{
+			name:     "error only: string description",
+			expected: &tracepb.Status{Code: 2},
+			attributes: map[string]string{
+				"error": "a description",
+			},
+		},
+
+		{
+			name:     "error only: true",
+			expected: &tracepb.Status{Code: 2},
+			attributes: map[string]string{
+				"error": "true",
+			},
+		},
+
+		{
+			name:     "error only: false",
+			expected: &tracepb.Status{Code: 2},
+			attributes: map[string]string{
+				"error": "false",
```

Comment:
This case is still weird to me, I looked through our data and we do have spans with this being reported 🙄 

How do you feel about treating a boolean attribute with a value `false` as OK?

Reply:
I looked into this more, I don't think we need to make any changes. 

Originally I was looking at our exported data in honeycomb where this was accidentally reported as a boolean instead of a string, causing some type conversion problems.